### PR TITLE
chore: clean up contract call sanity precheck logic

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Refactor the logic for conditions to precheck for possible invalid contract call transactions
 
**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java [hunk 1]
```diff
@@ -255,6 +251,25 @@ private ResponseCodeEnum validateSemantics(final TransactionBody transactionBody
...
+
+        // Tokens are valid targets
+        if (entityAccess.isTokenAccount(target.toId().asEvmAddress())) {
+            return false;
+        }
+
+        // Check for possible lazy create
+        if (target.equals(EntityNum.MISSING_NUM)
+                && isOfEvmAddressSize(op.getContractID().getEvmAddress())
+                && op.getAmount() > 0) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java:266]: Should this be ?  
```suggestion
              (target.equals(EntityNum.MISSING_NUM)
                && isOfEvmAddressSize(op.getContractID().getEvmAddress())
                ||op.getAmount() > 0
```

Comment (lukelee-sl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java:266]: I suppose it depends on how broad or narrow this precheck should be.  I don't see any not to narrow the scope as you suggest.  Will change as suggested.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCallTransitionLogic.java:266]: My comment was just based on the check that existed in mono-service. Not sure if it needs to be the same also.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @lukelee-sl

**Comment by lukelee-sl** [comment]
> LGTM ! Thanks @lukelee-sl

Thank you for taking a look @Neeharika-Sompalli !

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**